src/main.cpp

```diff
@@ -105,44 +105,52 @@ static void deleteInputDevices(QMap<SDL_JoystickID, InputDevice *> *joysticks)
         qDebug() << "Legacy settings found";
-        if (!current.exists())
+        const QString successMessage = QString("Your original settings (previously stored in %1) "
+                                               "have been copied to "
+                                               "~/.config/antimicrox to ensure consistent naming across "
+                                               "entire project.\nIf you want you can "
+                                               "delete the original directory or leave it as it is.")
+                                           .arg(fileToCopy.canonicalFilePath());
+        const QString errorMessage = QString("Some problem with settings migration occurred.\nOriginal "
+                                             "configs are stored in %1 "
+                                             "but their new location is ~/.config/antimicrox.\n"
+                                             "You can migrate manually by renaming old directory and "
+                                             "renaming file to antimicrox_settings.ini.")
+                                         .arg(fileToCopy.canonicalFilePath());
+
+        QMessageBox msgBox;
+        if (copySuccess)
         {
-            if (QFile::copy(PadderCommon::configLegacyFilePath(), PadderCommon::configFilePath()))
-            {
-                qDebug() << "Legacy antimicroX settings copied";
-                QMessageBox msgBox;
-                msgBox.setText("Your original settings (previously stored in "
-                               "~/.config/antimicroX) have been copied to "
-                               "~/.config/antimicrox to ensure consistent naming across "
-                               "entire project, if you want you can "
-                               "delete original directory or leave it as it is");
-                msgBox.exec();
-            } else
-            {
-                qDebug() << "Problem with importing antimicroX settings from: " << PadderCommon::configLegacyFilePath()
-                         << " to: " << PadderCommon::configFilePath();
-                QMessageBox msgBox;
-                msgBox.setText("Some problem with settings migration occurred.\nOriginal "
-                               "configs are stored in ~/.config/antimicroX, "
-                               "but their new location is ~/.config/antimicrox\nYou can "
-                               "do it manually by renaming old directory and "
-                               "renaming file antimicroX_settings.ini to "
-                               "antimicrox_settings.ini");
-                msgBox.exec();
-            }
+            qDebug() << "Legacy settings copied";
+            msgBox.setText(successMessage);
+        } else
+        {
+            qDebug() << "Problem with importing settings from: " << fileToCopy.canonicalFilePath()
```

Comment:
I think we could use here qWarning()

Reply:
OK. It was your code I didn't change it :laughing:

Reply:
Fixed